client/settings/digital-wallets/index.js

```diff
@@ -15,27 +14,33 @@ import {
 import interpolateComponents from 'interpolate-components';
 import { getPaymentMethodSettingsUrl } from '../../utils';
 
+/**
+ * Internal dependencies
+ */
+import { useDigitalWalletsSettings, useDigitalWalletsLocations } from 'data';
+
 const DigitalWallets = () => {
-	const [ isEnabled, setIsEnabled ] = useState( false );
-	const [ sectionsStatus, setSectionsStatus ] = useState( {
-		checkout: true,
-		productPage: true,
-		cart: true,
-	} );
+	const {
+		isDigitalWalletsEnabled,
+		updateIsDigitalWalletsEnabled,
+	} = useDigitalWalletsSettings();
+	const {
+		digitalWalletsLocations,
+		updateDigitalWalletsLocations,
+	} = useDigitalWalletsLocations();
 
-	const makeHandleSectionStatusChange = ( section ) => ( status ) => {
-		setSectionsStatus( ( oldStatuses ) => ( {
-			...oldStatuses,
-			[ section ]: status,
-		} ) );
+	const locationChangeHandler = ( location ) => ( status ) => {
```

Comment:
(⛏️ ) Looks like this is a curry pattern - so the `locationChangeHandler` is more of a factory for the location change handler. What do you think about calling this `makeLocationChangeHandler` or `makeHandleLocationChange`?